synapse/handlers/federation.py

```diff
@@ -101,6 +102,8 @@
 
 logger = logging.getLogger(__name__)
 
+soft_failed_event_counter = Counter("synapse_handlers_federation_soft_failed_event", "")
```

Comment:
This metric name is based off of the following counters already in the codebase. It looks like there is already inconsistency in the plurality. It makes sense that it matches the file name `synapse/handlers/federation.py`

 - [`synapse_handlers_appservice_events_processed`](https://github.com/matrix-org/synapse/blob/aec2cf1c9832eac46e1d94a39e75490b02320555/synapse/handlers/appservice.py#L44)
 - [`synapse_handler_presence_notified_presence`](https://github.com/matrix-org/synapse/blob/aec2cf1c9832eac46e1d94a39e75490b02320555/synapse/handlers/presence.py#L75)
  - [`synapse_storage_events_persisted_events`](https://github.com/matrix-org/synapse/blob/aec2cf1c9832eac46e1d94a39e75490b02320555/synapse/storage/databases/main/events.py#L58)
   - [`synapse_storage_events_state_delta_single_event`](https://github.com/matrix-org/synapse/blob/aec2cf1c9832eac46e1d94a39e75490b02320555/synapse/storage/persist_events.py#L51)

Reply:
> synapse_handler_presence_notified_presence (plurality different)

I'm not sure what the plural of that would be. `synapse_handler_presence_notified_presences`, maybe.

Counters should generally be plural, and end in `_total`, per https://prometheus.io/docs/practices/naming - `synapse_handlers_federation_soft_failed_events_total` in this case. (Actually the prometheus library will add `_total` for you if you don't add it.)

Some background on the naming of our existing metrics: once upon a time, someone thought it would be a good idea to name them according to the python module they were defined in. I don't know why they thought that, but I don't think it was a good idea, and I don't necessarily think we need to stick slavishly to it. 

I think I would go with `synapse_federation_soft_failed_events_total` here.

Reply:
Thanks for the docs and context here! Updated to `synapse_federation_soft_failed_events_total`